torch/csrc/jit/codegen/cuda/type.cpp

```diff
@@ -87,6 +87,9 @@ ValType promote_type(const ValType& t1, const ValType& t2) {
       (t1 == ValType::Scalar || t1 == ValType::NamedScalar)) {
     return ValType::Scalar;
   }
+  if (t1 == ValType::NamedScalar && t2 == ValType::NamedScalar) {
+    return ValType::Scalar;
+  }
```

Comment:
Looks like `promote_type` should just return `ValType::Scalar` whenever none of the inputs is `TensorView`.

Reply:
Likely, but want to keep this narrow and assert on any unexpected val types sent as someone could send something like TensorIndex which wouldn't make any sense here.